lightning/src/routing/network_graph.rs

```diff
@@ -125,10 +125,10 @@ impl<C: Deref + Sync + Send, L: Deref + Sync + Send> RoutingMessageHandler for N
 				let _ = self.network_graph.write().unwrap().update_channel(msg, Some(&self.secp_ctx));
 			},
 			&msgs::HTLCFailChannelUpdate::ChannelClosed { ref short_channel_id, ref is_permanent } => {
-				self.network_graph.write().unwrap().close_channel_from_update(short_channel_id, &is_permanent);
+				self.network_graph.write().unwrap().close_channel_from_update(*short_channel_id, *is_permanent);
```

Comment:
I think you can remove the `ref`'s from short_channel_id and is_permanent, then remove the dereferences (and probably the same below)